packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -0,0 +1,102 @@
+import {Count, DataObject, Entity, Filter, Options, Where} from '../..';
+
+/**
+ * CRUD operations for a target repository of a HasManyThrough relation
+ */
+export interface HasManyThroughRepository<
+  Target extends Entity,
+  TargetID,
+  Through extends Entity
+> {
+  /**
+   * Create a target model instance
+   * @param targetModelData - The target model data
+   * @param throughModelData - The through model data
+   * @param options - Options for the operation
+   * @param throughOptions - Options passed to create through
+   * @returns A promise which resolves to the newly created target model instance
+   */
+  create(
+    targetModelData: DataObject<Target>,
+    throughModelData?: DataObject<Through>,
+    options?: Options,
+    throughOptions?: Options,
+  ): Promise<Target>;
```

Comment:
I find this signature as too easy to misuse. Consider the following call:

```ts
await repo.create({name: 'Miroslav'}, options);
```

The pattern look very similar to what we do with other repository classes (e.g. `HasManyRepository`), but it is actually an error because the second argument is "through data", not "options".

I am proposing to move `throughModelData` and `throughOptions` into `options` argument.

```ts
export interface HasManyThroughRepository<
   Target extends Entity,
   TargetID,
   Through extends Entity
 > {

  create(
    targetModelData: DataObject<Target>,
    options: Options & {throughData?: DataObject<Through>; throughOptions?: Options;},
  ): Promise<Target>;
}
```